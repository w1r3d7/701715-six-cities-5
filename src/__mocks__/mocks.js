export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const FilterType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const offers = [{
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13
    }
  },
  title: `The house among olive `,
  rating: 3.8,
  type: `house`,
  bedrooms: 3,
  price: 493,
  description: `The house among olive `,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16
  },
  id: 1,
  bedroomsCount: 3,
  facilities: [
    `Laptop friendly workspace`,
    `Washer`,
    `Towels`,
    `Fridge`,
    `Air conditioning`,
    `Baby seat`,
    `Breakfast`
  ],
  photosUrl: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
  ],
  isPremium: false,
  isInBookmark: false,
  hostName: `Angelina`,
  hostDescription: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  hostAvatar: `img/avatar-angelina.jpg`,
  isHostPremium: true,
  maxCapacity: 4,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  title: `Amazing and Extremely Central Flat`,
  rating: 4.3,
  type: `apartment`,
  bedrooms: 3,
  price: 323,
  description: `Amazing and Extremely Central Flat`,
  location: {
    latitude: 50.839557,
    longitude: 4.346697,
    zoom: 16
  },
  id: 2,
  bedroomsCount: 3,
  facilities: [
    `Towels`,
    `Baby seat`,
    `Laptop friendly workspace`,
    `Dishwasher`,
    `Washing machine`,
    `Air conditioning`,
    `Washer`,
    `Coffee machine`,
    `Breakfast`,
    `Fridge`
  ],
  photosUrl: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`
  ],
  isPremium: false,
  isInBookmark: false,
  hostName: `Angelina`,
  hostDescription: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  hostAvatar: `img/avatar-angelina.jpg`,
  isHostPremium: true,
  maxCapacity: 8,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  title: `The house among olive `,
  rating: 4.6,
  type: `apartment`,
  bedrooms: 4,
  price: 218,
  description: `The house among olive `,
  location: {
    latitude: 50.865556999999995,
    longitude: 4.371696999999999,
    zoom: 16
  },
  id: 3,
  bedroomsCount: 4,
  facilities: [
    `Washer`,
    `Laptop friendly workspace`,
    `Breakfast`,
    `Air conditioning`,
    `Baby seat`
  ],
  photosUrl: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  isPremium: false,
  isInBookmark: false,
  hostName: `Angelina`,
  hostDescription: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
  hostAvatar: `img/avatar-angelina.jpg`,
  isHostPremium: true,
  maxCapacity: 7,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
}
];


export const reviews = [{
  "id": 1,
  "rating": 4,
  "date": `2020-11-15T13:58:19.494Z`,
  "reviewText": `421421`,
  "photoUrl": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
  "name": `xyz`,
  "isUserPremium": false,
  "userId": 1
}, {
  "id": 2,
  "rating": 3,
  "date": `2020-10-15T13:58:19.494Z`,
  "reviewText": `1111`,
  "photoUrl": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/4.jpg`,
  "name": `xyza`,
  "isUserPremium": true,
  "userId": 2
}
];
