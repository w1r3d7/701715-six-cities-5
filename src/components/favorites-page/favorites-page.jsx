import React, {useEffect} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import AppHeader from '../app-header/app-header';
import AppFooter from '../app-footer/app-footer';
import FavoritesContainer from '../favorites-container/favorites-container';

import {fetchFavoriteOffers} from '../../store/data/api-actions';
import {OFFER_PROP_TYPES} from '../../types.js';
import {getFavoriteLoadingStatus, getFavoriteOffers, getFavoriteOffersError} from '../../store/selectors';

const FavoritesPage = ({
  favoriteLoadingStatus,
  favoriteOffers,
  fetchFavoriteOffersAction,
  favoriteOffersError
}) => {

  useEffect(() => {
    fetchFavoriteOffersAction();
  }, []);

  return (
    <div className="page">
      <AppHeader />
      <FavoritesContainer
        isLoaded={favoriteLoadingStatus}
        favoriteOffers={favoriteOffers}
        error={favoriteOffersError}
      />
      <AppFooter />
    </div>
  );
};

FavoritesPage.propTypes = {
  favoriteOffers: PropTypes.arrayOf(
      PropTypes.shape(OFFER_PROP_TYPES).isRequired
  ),
  favoriteLoadingStatus: PropTypes.bool.isRequired,
  fetchFavoriteOffersAction: PropTypes.func.isRequired,
  favoriteOffersError: PropTypes.string
};

const mapDispatchToProps = (dispatch) => ({
  fetchFavoriteOffersAction: () => dispatch(fetchFavoriteOffers())
});

const mapStateToProps = (state) => ({
  favoriteOffers: getFavoriteOffers(state),
  favoriteLoadingStatus: getFavoriteLoadingStatus(state),
  favoriteOffersError: getFavoriteOffersError(state),
});

export {FavoritesPage};
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesPage);
