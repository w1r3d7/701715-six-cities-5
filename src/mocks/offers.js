const getRating = () => {
  return String((Math.random() * 5).toFixed(2));
};

const getRandomItemsCount = (array) => {
  const shuffled = array.sort(() => 0.5 - Math.random());
  return shuffled.slice(0, Math.floor(Math.random() * array.length));
};

const reviews = [
  {
    name: `Oleg`,
    id: 11,
    photoUrl: `/img/avatar-max.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
  {
    name: `Petr`,
    id: 21,
    photoUrl: `/img/avatar-max.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
  {
    name: `Max`,
    id: 31,
    photoUrl: `/img/avatar-max.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
  {
    name: `Angelina`,
    id: 41,
    photoUrl: `/img/avatar-angelina.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
];

export default [
  {
    id: 1,
    city: `Amsterdam`,
    photosUrl: [`/img/apartment-01.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `Apartment`,
    price: 120,
    description: `Beautiful & luxurious studio at great location`,
    isPremium: true,
    isInBookmark: false,
    facilities: [
      `Wi-Fi`,
      `Towels`,
      `Washing machine`,
      `Heating`,
    ],
    rating: getRating(),
    bedroomsCount: 3,
    maxCapacity: 4,
    hostName: `Angelina`,
    hostAvatar: `/img/avatar-angelina.jpg`,
    isHostPremium: true,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolor eius et facilis fuga mollitia
      praesentium quasi qui quibusdam quo recusandae repellat, repellendus rerum saepe sapiente sequi unde velit vero
      voluptatem.`,
    reviews: getRandomItemsCount(reviews),
  },
  {
    id: 2,
    city: `Amsterdam`,
    photosUrl: [`/img/apartment-02.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `House`,
    price: 40,
    description: `Great location`,
    isPremium: true,
    isInBookmark: true,
    facilities: [
      `Wi-Fi`,
      `Towels`,
      `Washing machine`,
      `Heating`,
    ],
    rating: getRating(),
    bedroomsCount: 1,
    maxCapacity: 2,
    hostName: `Igor`,
    hostAvatar: `/img/avatar-max.jpg`,
    isHostPremium: false,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Blanditiis doloremque in iure
      laborum magnam placeat quasi quidem quo! Assumenda error fuga iste laborum molestiae nam odit quia repellendus sed
      soluta.`,
    reviews: getRandomItemsCount(reviews),
  },
  {
    id: 3,
    city: `Amsterdam`,
    photosUrl: [`/img/apartment-03.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `Villa`,
    price: 4000,
    description: `Nice, cozy, warm big bed apartment`,
    isPremium: false,
    isInBookmark: false,
    facilities: [
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
    ],
    rating: getRating(),
    bedroomsCount: 4,
    maxCapacity: 5,
    hostName: `Oleg`,
    hostAvatar: `/img/avatar-max.jpg`,
    isHostPremium: true,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Blanditiis doloremque in iure
      laborum magnam placeat quasi quidem quo! Assumenda error fuga iste laborum molestiae nam odit quia repellendus sed
      soluta.`,
    reviews: getRandomItemsCount(reviews),
  },
  {
    id: 4,
    city: `Amsterdam`,
    photosUrl: [`/img/room.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `Private Room`,
    price: 4,
    description: `Wood and stone place`,
    isPremium: false,
    isInBookmark: true,
    facilities: [
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    rating: getRating(),
    bedroomsCount: 1,
    maxCapacity: 1,
    hostName: `Oleg`,
    hostAvatar: `/img/avatar-max.jpg`,
    isHostPremium: false,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Blanditiis doloremque in iure
      laborum magnam placeat quasi quidem quo! Assumenda error fuga iste laborum molestiae nam odit quia repellendus sed
      soluta.`,
    reviews: getRandomItemsCount(reviews),
  },
];
