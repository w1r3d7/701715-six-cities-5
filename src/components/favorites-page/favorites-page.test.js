import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import {FavoritesPage} from './favorites-page';

import {offers} from '../../__mocks__/mocks';

const mockStore = configureStore();

test(`FavoritesPage render correctly`, () => {
  const initialState = {
    DATA: {
      favoriteOffers: offers,
    },
    USER: {
      authorizationStatus: `NO_AUTH`,
      info: {}
    }
  };

  const store = mockStore(initialState);

  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <FavoritesPage
              favoriteOffers={offers}
              fetchFavoriteOffersAction={() => {}}
              favoriteLoadingStatus={true}
            />
          </BrowserRouter>
        </Provider>

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
