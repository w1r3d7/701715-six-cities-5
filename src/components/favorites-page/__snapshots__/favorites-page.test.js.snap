// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Cologne
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/1"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        493
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button "
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "76%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Brussels
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/2"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        323
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button "
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "86%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/2"
                      onClick={[Function]}
                    >
                      Amazing and Extremely Central Flat
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    apartment
                  </p>
                </div>
              </article>
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/3"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        218
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button "
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "92%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/3"
                      onClick={[Function]}
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
